### ISSUE TITLE
Enabler: Update Composer to v2

### ISSUE THREAD

**Issue by blairlearn** [NCIOCPL/cgov-digital-platform#3002]
# Update Composer to v2.

TL;DR; The "upgrade" to composer 2 for our composer.json and composer.lock is really just updating composer plugins as seen in #3178. The steps below for upgrading the tooling still stand as needed, but once #3178 is in you can use composer 2 with our project. (Even our commands work)

------------------------
>**_WARNING:_** This update is not as simple as updating composer. it requires an update to BLT 12 as well. So this will most likely have a snowball effect requiring other things.

The composer tool is now on version 2, with version 1 (the version we're currently using) at end of life.

### High Level Steps
These steps do not detail how to upgrade composer, but what needs to be done as part of this ticket.
- [ ] Update our composer.json/composer.lock to be usable by Composer 2. See https://www.drupal.org/docs/develop/using-composer/preparing-your-site-for-composer-2 for hints at what composer packages must be updated and what needs to be done.
- [x] Update our docker stack to install composer 2 by default
   - In `docker/web/Dockerfile` remove the `--1` switch when [installing composer](https://github.com/NCIOCPL/cgov-digital-platform/blob/develop/docker/web/Dockerfile#L131)
   - Remove [the container's global installation](https://github.com/NCIOCPL/cgov-digital-platform/blob/develop/docker/web/Dockerfile#L134) of the obsolete `hirak/prestissimo` package.
- [ ] Document, as well as possible, the steps developers need to follow to update their machines to use composer 2 from their command line.
   - !!!YOU MUST SUPPORT RUNNING **COMPOSER 1** FOR OLDER BRANCHES AND FORKS!!!
- [ ] Update pipelines configuration to use Composer 2

**Possible Issues**
* Our custom composer commands (cgov-init and cgov-clean)?
* The CTS API PHP Library package?

> **ESTIMATE** TBD

**Comment by bryanpizzillo** [comment]
Technically this is not a Drupal 9.x upgrade blocker, but a BLT 12.8 blocker, and we should be looking to move toward BLT 13 (the current version, and thus the one "supported").

**Comment by bryanpizzillo** [comment]
NOTE: `toggle:modules` is removed in BLT 13

**Comment by bryanpizzillo** [comment]
So more notes since we started. BLT 11.7.0 supports composer 2, so this task can be performed outside of any BLT upgrade to 12 and above. The notes for BLT 13 are valid, but not for this ticket.
